dali/dfu/dfuutil.cpp

```diff
@@ -876,20 +892,44 @@ class CFileCloner
             srcLFN.clearForeign();
             srcdali.setown(createINode(ep));
         }
+
         StringBuffer s;
-        Owned<IPropertyTree> ftree = fdir->getFileTree(srcLFN.get(), foreignuserdesc, srcdali, FOREIGN_DALI_TIMEOUT, GetFileTreeOpts::appendForeign);
-        if (!ftree.get())
-            throw MakeStringException(-1,"Source file %s could not be found in Dali %s",srcLFN.get(), getDaliEndPointStr(srcdali, s));
-        IPropertyTree *attsrc = ftree->queryPropTree("Attr");
-        if (!attsrc)
-            throw MakeStringException(-1,"Attributes for source file %s could not be found in Dali %s",srcLFN.get(), getDaliEndPointStr(srcdali, s));
+        Owned<IPropertyTree> ftree;
+        Owned<IPropertyTree> filePlane;
+        IPropertyTree *attsrc = nullptr;
+        if (remoteStorage || srcLFN.isRemote())
+        {
+            StringBuffer remoteLFN;
+            if (!srcLFN.isRemote())
+                remoteLFN.append("remote::").append(remoteStorage).append("::");
+            srcLFN.get(remoteLFN);
+
+            Owned<wsdfs::IDFSFile> dfsFile = wsdfs::lookupDFSFile(remoteLFN.str(), AccessMode::readSequential, INFINITE, wsdfs::keepAliveExpiryFrequency, foreignuserdesc);
+            IPropertyTree *tree = (dfsFile) ? dfsFile->queryFileMeta() : nullptr;
+            if (tree)
+                ftree.setown(tree->getPropTree("File"));
+            if (!ftree.get())
```

Comment:
would be slightly clearer if:
```
if (!dfsFile)
    throw makeStringExceptionV(-1,"Source file %s could not be found in Remote Storage", remoteLFN.str()); //remote scope already included in remoteLFN
ftree.setown(dfsFile->queryFilemeta()->getPropTree("File"));

```

Reply:
Agreed.  Wasn't sure those sections were guaranteed to be available.  Updated.